test/unit/BCoWPool.t.sol

```diff
@@ -130,3 +148,104 @@ contract BCoWPool_Unit_EnableTrading is BaseCoWPoolTest {
+    for (uint256 i = 0; i < TOKENS_AMOUNT; i++) {
+      vm.assume(tokens[i] != _token);
+    }
+    _;
+  }
+
+  function test_Revert_NonBoundBuyToken(address _otherToken) public assumeNotBoundToken(_otherToken) {
+    GPv2Order.Data memory order = correctOrder;
+    order.buyToken = IERC20(_otherToken);
+    vm.expectRevert(IBPool.BPool_TokenNotBound.selector);
+    bCoWPool.verify(order);
+  }
+
+  function test_Revert_NonBoundSellToken(address _otherToken) public assumeNotBoundToken(_otherToken) {
+    GPv2Order.Data memory order = correctOrder;
+    order.sellToken = IERC20(_otherToken);
+    vm.expectRevert(IBPool.BPool_TokenNotBound.selector);
+    bCoWPool.verify(order);
+  }
+
+  function test_Revert_LargeDurationOrder(uint256 _timeOffset) public {
+    _timeOffset = bound(_timeOffset, 5 minutes, type(uint16).max);
+    GPv2Order.Data memory order = correctOrder;
+    order.validTo = uint32(block.timestamp + _timeOffset);
+    vm.expectRevert(IBCoWPool.BCoWPool_OrderValidityTooLong.selector);
+    bCoWPool.verify(order);
+  }
+
+  function test_Revert_NonZeroFee(uint256 _fee) public {
+    _fee = bound(_fee, 1, type(uint256).max);
+    GPv2Order.Data memory order = correctOrder;
+    order.feeAmount = _fee;
+    vm.expectRevert(IBCoWPool.BCoWPool_FeeMustBeZero.selector);
+    bCoWPool.verify(order);
+  }
+
+  function test_Revert_InvalidOrderKind(bytes32 _orderKind) public {
+    vm.assume(_orderKind != GPv2Order.KIND_SELL);
+    GPv2Order.Data memory order = correctOrder;
+    order.kind = _orderKind;
+    vm.expectRevert(IBCoWPool.BCoWPool_InvalidOperation.selector);
+    bCoWPool.verify(order);
+  }
+
+  function test_Revert_InvalidBuyBalanceKind(bytes32 _balanceKind) public {
+    vm.assume(_balanceKind != GPv2Order.BALANCE_ERC20);
+    GPv2Order.Data memory order = correctOrder;
+    order = correctOrder;
+    order.buyTokenBalance = _balanceKind;
+    vm.expectRevert(IBCoWPool.BCoWPool_InvalidBalanceMarker.selector);
+    bCoWPool.verify(order);
+  }
+
+  function test_Revert_InvalidSellBalanceKind(bytes32 _balanceKind) public {
+    vm.assume(_balanceKind != GPv2Order.BALANCE_ERC20);
+    GPv2Order.Data memory order = correctOrder;
+    order = correctOrder;
+    order.sellTokenBalance = _balanceKind;
+    vm.expectRevert(IBCoWPool.BCoWPool_InvalidBalanceMarker.selector);
+    bCoWPool.verify(order);
+  }
+
+  function test_Revert_InsufficientReturn(uint256 _buyAmount, uint256 _offset) public {
+    _buyAmount = bound(_buyAmount, 1, type(uint128).max);
+    _offset = bound(_offset, 1, _buyAmount);
+    GPv2Order.Data memory order = correctOrder;
+    order.buyAmount = _buyAmount;
+    vm.mockCall(tokens[0], abi.encodePacked(IERC20.balanceOf.selector), abi.encode(1 ether));
+    vm.mockCall(tokens[1], abi.encodePacked(IERC20.balanceOf.selector), abi.encode(1 ether));
+    vm.mockCall(address(bCoWPool), abi.encodePacked(BMath.calcOutGivenIn.selector), abi.encode(_buyAmount - _offset));
```

Comment:
hmmm idk if i'm convinced by this, other option is to fuzz `amountIn, balanceIn, denormIn, balanceOut, denormOut`, i think i'd choose the latter

by overriding the internal call we're breaking the whole interaction imo